### PR TITLE
[extension/storage/filestorage] Document how to read files created by the `file_storage` extension

### PR DESCRIPTION
**Pull request by ycombinator**
**Description:** 

This PR adds documentation explaining how to read the contents of files created by the `file_storage` extension.

**Link to tracking Issue:**
* Related to #30970
* Follow up to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31886#discussion_r1542581347

### CODE REVIEW

**Comment by atoulme** [comment]
I'm not sure about this. Are you trying to achieve a particular functionality or achieve a use case such as troubleshooting?

**Comment by atoulme** [comment]
OK, I have read the associated issue. Would you mind framing this as a troubleshooting solution?

**Review by atoulme — APPROVED** [review]

File: extension/storage/filestorage/README.md [hunk 1]
```diff
@@ -133,3 +133,29 @@ The schedule for this feature gate is:
 - Removed three releases after `stable`.
 
 [unicode_chars]: https://en.wikipedia.org/wiki/List_of_Unicode_characters
+
+## Troubleshooting
+
+When troubleshooting components that use file storage, it is sometimes helpful to read the raw contents of
```
Comment (djaglowski) [inline extension/storage/filestorage/README.md:139]: Can we add a note that this strategy is subject to change.

The context here is that the underlying storage mechanism, aside from being a file, is not part of an API which we have ever guaranteed. Some suggestions have been made in the past to evaluate other key/value stores which may be more performant than the bbolt one we currently use. If we ever change, we would of course have to ensure migration is possible for some time, but ultimately we should be careful not to appear locked into the implementation.

**Review by tiffany76 — APPROVED** [review]
Looks good from a documentation perspective.

File: extension/storage/filestorage/README.md (hunk 1 above)
Comment (ycombinator) [reply extension/storage/filestorage/README.md]: Added in 4a32823b11.

**Review by tiffany76 — APPROVED** [review]
One small change to eliminate redundancy. Otherwise, LGTM.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**